zingolib/src/wallet/transaction_context.rs

```diff
@@ -44,503 +24,549 @@ impl TransactionContext {
             transaction_metadata_set,
         }
     }
+}
 
-    #[allow(clippy::too_many_arguments)]
-    async fn execute_bundlescans_internal(
-        &self,
-        transaction: &Transaction,
-        status: ConfirmationStatus,
-        block_time: u32,
-        is_outgoing_transaction: &mut bool,
-        outgoing_metadatas: &mut Vec<OutgoingTxData>,
-        arbitrary_memos_with_txids: &mut Vec<(ParsedMemo, TxId)>,
-        taddrs_set: &HashSet<String>,
-    ) {
-        //todo: investigate scanning all bundles simultaneously
+/// These functions are responsible for receiving a full Transaction and storing it, with a few major caveats.
+///  The first layer is CompactTransaction. see fn trial_decrypt_domain_specific_outputs
+///  in some cases, including send, read memos, discover outgoing transaction (mark change / scan for internal change), additional information and processing are required
+/// some of the steps in scan_full_tx are similar to or repeat steps in trial_ddso
+/// however, scan full tx is more limited than trial_ddso.
+/// scan_full_tx has no access to a transmitter. So it is incapable of **sending a request on a transmitter for another task to fetch a witnessed position**.
+/// unlike a viewkey wallet, a spendkey wallet MUST pass reread the block to find a witnessed position to pass to add_new_note. scan_full_tx cannot do this.
```

Comment:
"spendkey wallet MUST pass reread the block to find a witnessed position to pass to add_new_note."

extra copy of the work "pass"